homeassistant/components/elmo_alarm/__init__.py

```diff
@@ -0,0 +1,232 @@
+CONF_NAME = CONF_NAME
+CONF_ZONES = "zones"
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=5)
+
+SIGNAL_ZONE_CHANGED = "elmo_alarm.zone_changed"
+SIGNAL_INPUT_CHANGED = "elmo_alarm.input_changed"
+SIGNAL_ARMING_STATE_CHANGED = "elmo_alarm.arming_state_changed"
+
+INPUT_TYPE = "opening"
+ZONE_TYPE = "safety"
+
+ZONE_ARMED = 0
+ZONE_DISARMED = 1
+
+INPUT_ALERT = 1
+INPUT_WAIT = 0
+
+ZoneData = namedtuple("ZoneData", ["zone_id", "zone_name", "state"])
+InputData = namedtuple("InputData", ["input_id", "input_name", "state"])
+
+
+STATE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_ZONES): vol.All(cv.ensure_list),
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_VENDOR): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): vol.All(cv.time_period, cv.positive_timedelta),
+                vol.Required(CONF_STATES): vol.All(cv.ensure_list, [STATE_SCHEMA]),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the e-connect Elmo Alarm platform."""
+
+    conf = config[DOMAIN]
+
+    host = conf[CONF_HOST]
+    vendor = conf[CONF_VENDOR]
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    states = conf[CONF_STATES]
+    scan_interval = conf[CONF_SCAN_INTERVAL]
+
+    _LOGGER.warning(f"Scan Interval:{scan_interval}")
+
+    client = ElmoClientWrapper(host, vendor, username, password, states)
+    await client.update()
+
+    hass.data[DOMAIN] = client
+
+    hass.async_create_task(
+        async_load_platform(
+            hass,
+            "binary_sensor",
+            DOMAIN,
+            {"zones": client._zones, "inputs": client._inputs},
+            config,
+        )
+    )
+
+    hass.async_create_task(
+        async_load_platform(hass, "alarm_control_panel", DOMAIN, {}, config)
+    )
+
+    async def update():
+        _LOGGER.debug("Connecting to e-connect to retrieve states.")
+        await client.update()
+        async_dispatcher_send(hass, SIGNAL_ARMING_STATE_CHANGED, client._state)
+
+        for zone in client._zones:
+            async_dispatcher_send(hass, SIGNAL_ZONE_CHANGED, zone)
+
+        for inp in client._inputs:
+            async_dispatcher_send(hass, SIGNAL_INPUT_CHANGED, inp)
+
+        async_call_later(
+            hass,
+            scan_interval.total_seconds(),
+            lambda _: hass.async_create_task(update()),
+        )
+
+    await update()
+
+    return True
+
+
+class ElmoClientWrapper(ElmoClient):
+    """Wrapping the Elmo client class to adapt it to home assistant"""
+
+    def __init__(self, host, vendor, username, password, states_config):
```

Comment:
Still missing the docstring for the __init__ method.

Reply:
Sorry, added.